genesis/public/core/reference.go

```diff
@@ -106,44 +106,47 @@ func (rd *referenceDomain) ResolveReference(record string) (*object.Reference, e
 }
 
 type referenceDomainProxy struct {
```

Comment:
referenceDomainProxy is not an implementation of ReferenceDomain interface now, but it should be